apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml

```diff
@@ -168,6 +168,13 @@
             <version>${mockito.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <!-- Configures mockito to allow mocking of final classes and other features. -->
```

Comment:
Can you make a note here as well that this was added to resolve the illegal reflection warning?

I'm curious as well as to how/why this resolves it. I did a search and I see other references to this but I didn't really figure out what it is doing that fixes it. Anything found from your side that answers that question?

Reply:
If you look in the `mockito-inline` jar, it's just a couple ascii files, as the only thing it's doing is providing a one-line configuration to tell it which MockMaker implementation to use.  It's the difference of it using the `ByteBuddyMockMaker`, which results in the illegal reflection issue, versus the `InlineByteBuddyMockMaker` which doesn't, but as to how the implementations differ to cause the illegal reflection access or not, I'm not exactly sure.

https://github.com/hustwei/mockito/blob/master/src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

In this PR's code's case, it was when Mockito changed the access level for the `protected volatile InputStream in` property of `FilterInputStream` that was throwing the illegal reflection.  Using the inline version, it's able to do this without causing an illegal reflection action.

Reply:
oh and I'll add the extra note

Reply:
Thanks. This is helpful. Would you mind putting this additional information in the PR description as well?